lib/_http_server.js

```diff
@@ -378,9 +378,6 @@ function writeHead(statusCode, reason, obj) {
         if (k) this.setHeader(k, obj[k]);
       }
     }
-    if (k === undefined && this._header) {
```

Comment:
Why did you remove this?

Reply:
it never enters that condition since https://github.com/nodejs/node/pull/45508.
if(this._header) is true it will throw before